src/sentry/api/event_search.py

```diff
@@ -817,6 +831,8 @@ def visit_function_args(self, node, children):
 
     def visit_search_key(self, node, children):
         key = children[0]
+        if self.config.allowed_keys and key not in self.config.allowed_keys:
+            raise InvalidSearchQuery("Invalid key for this search")
```

Comment:
this is a huge pita to do probably, but I wonder if it would be nice not to produce an error, but instead just translate it into free text?

Reply:
Hmm... I think I'd rather error so it's clear what's supported or not, just in case people are pasting things between searches.

Reply:
Ideally the frontend would handle this as well with it's syntax highlighting.

Though in that case, we can easily mark some key:value filter as being 'invalid'.

Ok, I think this is fine.

Reply:
I'm fine to revisit this later too if we feel like it was a bad choice

Reply:
Yeah, sgtm. I think we'll know more once we build out the frontend experience for searching releases.